jumanji/environments/combinatorial/routing/rb_board_generator_v1.py

```diff
@@ -0,0 +1,319 @@
+            bottom_empty = True
+        if (y==self.dim[1]-1) or self.layout[x, y+1]==0:
+            right_empty = True
+        # Check for full corner cells
+        if (x > 0) and (y > 0) and self.layout[x-1, y-1]:
+            upleft_full = True
+        if (x > 0) and (y < self.dim[1]-1) and self.layout[x-1, y+1]:
+            upright_full = True
+        if (x < self.dim[0]-1) and (y > 0) and self.layout[x+1, y-1]:
+            botleft_full = True
+        if (x < self.dim[0]-1) and (y < self.dim[1] -1) and self.layout[x+1, y+1]:
+            botright_full = True
+        # Check if it's a corner cell we can remove
+        # If two neighboring adjacent cells are unconneted, it's a corner
+        # If the opposit diagonal is connected, this corner is redundant.
+        if (upper_empty and left_empty and botright_full)\
+            or (upper_empty and right_empty and botleft_full)\
+            or (bottom_empty and left_empty and upright_full)\
+            or (bottom_empty and right_empty and upleft_full):
+            return True
+        else:
+            return False
+    """
+
+
+def remove_connections(layout_in: np.ndarray) -> np.ndarray:
+    """
+    Return a copy of the input board layout with the connecting wires zeroed out.
+
+    ARGS:
+        layout_in (2D numpy array of ints) : layout specifying heads, targets, and connectors
+    RETURNS:
+        layout_out (2D numpy array of ints) : same as layout_in with only heads and targets
+    """
+    layout_out = []
+    for row_in in layout_in:
+        # Zero out any element that is a connecting wire
+        row_out = [i * int(i%3 != WIRE) for i in row_in]
+        layout_out.append(row_out)
+    return np.array(layout_out)
+
+
+def board_generator(x_dim: int, y_dim: int, target_wires: int = None) -> (np.ndarray, np.ndarray, int):
+    """ Generate a circuit board of the specified size and number of wires
+
+    The circuit board will be of dimensions x_dim by y_dim.
+    The generator will attempt to add the number of wires specified by target_wires.
+    The wiring head/target pairs will be generated randomly.
+    If num_wires is not specified (or if it is larger than the number of
+    wires that the random process is able to fit), then the generator sill fit as
+    many wires as possible.
+	
+    The coding of the output boards is outlined in the Board class definition.
+
+    Args:
+        x_dim, y_dim (int, int) : Dimensions of the board
+	    num_wires (int) : Number of wiring pairs to attempt.
+	        Default (None) => Fit as many wiring pairs as possible.
+
+    Returns:
+        board_training (numpy array of 2D integer tuples (x_dim, y_dim)) 
+            Board design with head/target wiring ports marked
+        board_solution (numpy array of 2D integer tuples (x_dim, y_dim)) 
+            Board design with connecting wires and head/target ports marked
+	    num_wires_out (int) : number of wires in the output board design
```

Comment:
correct type for an np array is np.ndarray and afaik there isnt a good way to specify the size. Worth looking into.
Missing fullstops at end of sentences